app/helpers/application_helper.rb

```diff
@@ -28,7 +28,9 @@ def notices_alerts
   end
 
   def invite_btn(user)
-    return if current_user.check_if_my_friend(user) || current_user == user
+    bonds = Bond.where(friend: current_user, user: user, confirmed: false)
+    friend = bonds.map(&:user)
+    return if current_user == user || friend.include?(user) || current_user.check_if_my_friend(user)
 
     button_to('Invite to Friendship', invite_path(friend_id: user.id), method: :post)
```

Comment:
you should never use the `where` method, or any other query-building methods like this, outside of the model class itself, Take a look at this article for more Ruby on Rails best practices: [Buggy Rails Code: The 10 Most Common Mistakes That Rails Developers Make](https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make)